### PR TITLE
PaginatedDataTable improvements

### PR DESCRIPTION
**Pull request by Hixie**
- slightly improved assert message when row cell counts don't match column count.
- more breadcrumbs in API documentation. more documentation in general.
- added more documentation for the direction of the "ascending" arrow.
- two samples for PaginatedDataTable.
- make PaginatedDataTable support hot reloading across changes to the number of columns.
- introduce matrix3MoreOrLessEquals. An earlier version of this PR used it in tests, but eventually it was not needed. The function seems useful to keep though.

### CODE REVIEW

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 1]
```diff
@@ -301,6 +320,12 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
     }
   }
 
+  @override
+  void reassemble() {
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/paginated_data_table.dart:324]: Overriding this method is pretty rare (I think Image is the only other widget that does it). A comment explaining the what and why would be helpful.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/paginated_data_table.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/material/paginated_data_table.dart:324]: done

Comment (HansMuller) [reply packages/flutter/lib/src/material/paginated_data_table.dart:324]: NICE, more interesting than I expected!

**Pull request merged**